matching/src/main/scala/org/kframework/backend/llvm/matching/Heuristics.scala

```diff
@@ -0,0 +1,61 @@
+package org.kframework.backend.llvm.matching
+
+import org.kframework.parser.kore.SymbolOrAlias
+import org.kframework.backend.llvm.matching.pattern._
+
+trait Heuristic {
+  def scoreAs[T](f: Fringe, c: Clause, key: Option[Pattern[Option[Occurrence]]], isEmpty: Boolean, pat: Pattern[T]): Double
```

Comment:
let's remove the arguments after isEmpty and add a typed argument to the beginning with the type of the pattern. e.g., for this line it would be pat: AsP[T].